docs/output_file_organization.rst

```diff
@@ -0,0 +1,59 @@
+Organization of TAPA Output Files
+==========================================
+
+Based on different input options, `tapac` will generate multiple potential output files.
+
+The XO Object
+------------------
+
+In the most basic case, ``tapac`` takes in the source code and generate an XO object with the suffix ``.xo``. The XO object will be passed to the Vitis ``v++`` compiler as the input for bitstream generation.
+
+- Actually, the XO object is a ``.zip`` file with a different name.
+
+- If you ``unzip`` the XO object, you will see (1) all the RTLs corresponding to the C++ source code and (2) configuration files with metadata about the design.
+
+- If you like, you could modify the contents of the XO file. Simply unzip the XO object and modify the RTL as you want, then zip it back and passing it to ``v++``.
+
+The XXX_generate_bitstream.sh script
+-------------------------------------------------
+
+``tapac`` will automatically generate this script, which will invokes ``v++`` to compile the ``.xo`` object into the final bitstream. The parameters in the script will be matching the options passed to `tapac`.
+
+- By default, the script will use the ``Explore`` option for ``opt_design``, ``phys_opt_design``, ``route_design`` and use the ``EarlyBlockPlacement`` option for ``place_design``.
+
+- The default frequency target is 300 MHz. For HBM applications, you could edit the script if a lower frequency is enough to saturate the HBM bandwidth. We usually set the target slightly above what could be actually achieved.
```

Comment:
I use 300 only if --clock-period is not specified, in which case the default of v++ is 300

Reply:
I mean "edit the script if a lower frequency is enough to saturate the HBM bandwidth." is not a good idea. Users should change `--clock-period` if they want to lower the target frequency, not edit the generated script.

Reply:
It's common that HLS target freq is different from P&R. For example, with 300 MHz target in HLS, perhaps more pipelining is scheduled; while a lower target for Vivado (say 250 MHz) allows more time spent on HBM clock optimization. In my experience, it achieves the best result if the Vivado target is slightly above the physical limit.

Reply:
In this case, why not specify the HLS target as 250MHz as well, to allow potentially less resource usage? Is it because the HLS timing model often underestimates the delay?